src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -333,16 +335,16 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
     const workspaceLoaded = !!workspace
 
     const { runtimes, refreshRuntimes, persistentDisks, appDataDisks } = useCloudEnvironmentPolling(googleProject, workspace)
-    const { apps, refreshApps } = useAppPolling(googleProject, name)
+    const { apps, refreshApps } = useAppPolling(googleProject, name, workspace)
```

Comment:
I think that it would work to move the `useAppPolling` call down past line 339, so that it is only executed if the workspace is loaded. Then you wouldn't need to check !!workspace in useAppPolling.

Reply:
Yeah that can be done. I didn't touch it because I wasn't sure if it would affect any other logic in here.